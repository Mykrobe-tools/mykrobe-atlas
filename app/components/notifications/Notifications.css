.wrap {
  width: 100%;
}

.notification {
  width: 100%;
}
