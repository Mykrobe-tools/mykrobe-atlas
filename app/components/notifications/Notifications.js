/* @flow */

import React, { Component, PropTypes } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as NotificationActions from '../../actions/NotificationActions';

import Notification from './Notification';
import styles from './Notifications.css';

export class Notifications extends Component {

  constructor(props: Object) {
    super(props);
  }

  onClick(id: Number) {
    const {hideNotification} = this.props;
    hideNotification(id);
  }

  render() {
    const {notifications} = this.props;
    return (
      <div className={styles.wrap}>
        {notifications.map(notification => {
          const {id, content, category} = notification;
          return (
            <div key={id}>
              <Notification
                content={content}
                category={category.toLowerCase()}
                id={id}
                onClick={event => this.onClick(id)} />
            </div>
          )
        })}
      </div>
    );
  }
}

Notifications.propTypes = {
  notifications: PropTypes.array
}

function mapStateToProps(state) {
  return {
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    hideNotification: NotificationActions.hideNotification
  }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Notifications);
