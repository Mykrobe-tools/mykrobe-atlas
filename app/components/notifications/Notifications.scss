@import '~styles/colors.global';

.container {
}
