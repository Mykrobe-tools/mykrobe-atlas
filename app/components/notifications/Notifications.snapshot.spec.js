import React from 'react';
import renderer from 'react-test-renderer';
import {Notifications} from './Notifications';

describe('Notifications component snapshot', () => {
  const notifications = [
    {
      id: 0,
      content: "Lorem",
      category: "success"
    },
    {
      id: 1,
      content: "Ipsum",
      category: "error"
    }
  ];

  it('renders correctly', () => {
    const component = renderer.create(
      <Notifications
        notifications={notifications} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
