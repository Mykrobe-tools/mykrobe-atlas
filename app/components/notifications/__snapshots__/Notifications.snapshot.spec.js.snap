exports[`Notifications component snapshot renders correctly 1`] = `
<div
  className="wrap">
  <div
    className="notification">
    <a
      className="success"
      onClick={[Function]}>
      <p
        className="content">
        <span
          className="icon">
          <i
            className="fa fa-check-circle" />
        </span>
        Lorem
      </p>
      <span
        className="close">
        <i
          className="fa fa-times-circle" />
      </span>
    </a>
  </div>
  <div
    className="notification">
    <a
      className="error"
      onClick={[Function]}>
      <p
        className="content">
        <span
          className="icon">
          <i
            className="fa fa-exclamation-circle" />
        </span>
        Ipsum
      </p>
      <span
        className="close">
        <i
          className="fa fa-times-circle" />
      </span>
    </a>
  </div>
</div>
`;
